Amethyst/Managers/ScreenManager.swift

```diff
@@ -367,6 +368,26 @@ final class ScreenManager<Delegate: ScreenManagerDelegate>: NSObject, Codable {
     @objc func hideLayoutHUD(_ sender: AnyObject) {
         layoutNameWindowController.close()
     }
+
+    func toggleFullscreen() {
+        guard let currentLayout = currentLayout?.layoutKey else {
+            return
+        }
+
+        if currentLayout == "fullscreen" {
+            previousLayout = previousLayout ?? layouts.first(where: {$0.layoutKey != "fullscreen"})?.layoutKey
+
+            if previousLayout == nil {
+                return
+            }
+
+            selectLayout(previousLayout!)
```

Comment:
Why not use a `guard let` instead of force unwrapping?

Reply:
Don't have a good reason for that other than I assumed using 

`guard let previousLayout = previousLayout ?? layouts.first(where: {$0.layoutKey != "fullscreen"})?.layoutKey else {
                return
            }` 

would create a local variable rather than asign to the previousLayout in the outer scope.

As I mentioned, not at all familiar with swift. Happy to change it

Reply:
I would rewrite this as below to reduce the nesting and remove the force unwrapping. Swift is definitely not one of my primary languages either so maybe this can be done better.

```
    func toggleFullscreen() {
        guard let currentLayout = currentLayout?.layoutKey else {
            return
        }
        
        if (previousLayout != "fullscreen") {
            previousLayout = currentLayout
            selectLayout("fullscreen")
            return
        }
        
        guard let firstNonFullscreenLayout = layouts.first(where: {$0.layoutKey != "fullscreen"}),
              let previousLayout = previousLayout ?? firstNonFullscreenLayout else {
            return
        }

        selectLayout(previousLayout)
    }
```

Reply:
Based on your psuedo code I have this, which may be better than my original implementation

```
func toggleFullscreen() {
        guard let currentLayout = currentLayout?.layoutKey else {
            return
        }

        if currentLayout != "fullscreen" {
            previousLayout = currentLayout
            selectLayout("fullscreen")
            return
        }

        guard let layout = previousLayout ?? layouts.first(where: {$0.layoutKey != "fullscreen"})?.layoutKey else {
            return
        }

        previousLayout = layout
        selectLayout(layout)
    }
```